launcher/minecraft/mod/ResourcePack.cpp

```diff
@@ -0,0 +1,121 @@
+#include "ResourcePack.h"
+
+#include <QDebug>
+#include <QMap>
+#include <QRegularExpression>
+
+#include "Version.h"
+
+#include "minecraft/mod/tasks/LocalResourcePackParseTask.h"
+
+// Values taken from:
+// https://minecraft.fandom.com/wiki/Tutorials/Creating_a_resource_pack#Formatting_pack.mcmeta
+static const QMap<int, std::pair<Version, Version>> s_pack_format_versions = {
+    { 1, { Version("1.6.1"), Version("1.8.9") } }, { 2, { Version("1.9"), Version("1.10.2") } },
+    { 3, { Version("1.11"), Version("1.12.2") } }, { 4, { Version("1.13"), Version("1.14.4") } },
+    { 5, { Version("1.15"), Version("1.16.1") } }, { 6, { Version("1.16.2"), Version("1.16.5") } },
+    { 7, { Version("1.17"), Version("1.17.1") } }, { 8, { Version("1.18"), Version("1.18.2") } },
+    { 9, { Version("1.19"), Version("1.19.2") } },
```

Comment:
Isn't this gonna break any time a new version gets released?

Reply:
yea, but how else would we do it? the only other thing i can think of is using meta and updating it there, but it's harder :c

Reply:
Is there any version-dependant change we usd? Technically we could just ignore it

Reply:
oh, now i got the point
<img width="1517" alt="image" src="https://user-images.githubusercontent.com/83089242/188282354-f7d2eb04-0701-42fe-9514-43e74f6b6698.png">
issue is, when the pack format version is invalid, it ignores everything else, which isn't very future-proofing

Reply:
wait ignore what i said above, i was very dumdum

Reply:
i just noticed i added a test for whether a format is in the array when setting the pack format. maybe if it isn't in the array, we can just ignore it and not show the version info? or we could perhaps speculate of future versions, since mojang seems to be doing the formats be aligned to the major versions as of the last couple versions.

Reply:
i can also add an "invalid" or smt instead of '0' in case of not being able to read the thing properly, that would be better maybe

Reply:
maybe just what the pack format is (without the version range) or just "unrecognised"

Reply:
what i made now is:
if the pack format is 0 (the default), it says 'Unrecognized'
if the pack format is not 0, but isn't in that array, it'll only show the pack format number, without the versions

Reply:
It's fine now then